src/renderer/raycast_tool.c

```diff
@@ -0,0 +1,54 @@
+#include "renderer.h"
+
+t_colors	get_color(t_renderer *renderer, t_ivec *map, bool is_hit_y_side)
+{
+	const int		index = renderer->world.worldmap[map->y][map->x];
+	int				result;
+	const t_colors	colors[] = {
+		COLOR__YELLOW,
+		COLOR__RED,
+		COLOR__GREEN,
+		COLOR__BLUE,
+		COLOR__WHITE,
+	};
+
+	if (index > 4)
+		result = colors[0];
+	else
+		result = colors[index];
+	if (is_hit_y_side)
+		result /= 2;
+	return (result);
+}
```

Comment:
```suggestion
```

the function would not work under new map format, therefore it'd be better to remove the function and its use altogether

Reply:
done !